[export][refactor][1/n] Move dynamic shapes logic

angelayi opened the pull request:
1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114787
* #114768
* __->__ #114764

Differential Revision: [D51823962](https://our.internmc.facebook.com/intern/diff/D51823962)

angelayi committed:
[export][refactor] Split tracing logic

[ghstack-poisoned]

zhxchen17 commented on torch/export/exported_program.py:
```diff
@@ -604,3 +604,8 @@ def get_shape_env(gm):
     return range_constraints
+
+
+def combine_args_kwargs(args, kwargs):
```
do we need this function still? I think we should inline this like

```
(arg, kwargs or {})
```

angelayi replied:
It seems to be used in a lot of places 😅 https://www.internalfb.com/code/search?q=repo%3Afbcode%20combine_args_kwargs

zhxchen17 commented on torch/export/dynamic_shapes.py:
```diff
@@ -0,0 +1,568 @@
+import builtins
```
@tugsbayasgalan does this file have all the APIs?

zhxchen17 commented on torch/export/_tracer.py:
```diff
@@ -0,0 +1,615 @@
+import copy
```
@angelayi I'm nitpicking the name here. I prefer to merge this file with export/__init__.py, but it's also okay to keep it in separate file, better name to me is `_trace.py`

angelayi committed:
Update on "[export][refactor][1/n] Split tracing logic"

1/n of refactoring export code:
* Moved tracing logic in torch/_export/__init__.py to torch/export/_tracer.py
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py




[ghstack-poisoned]

ydwu4 reviewed (approved):
LGTM!

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51823962).

albanD commented on docs/source/conf.py:
```diff
@@ -694,6 +694,9 @@
     "tril_matrix_to_vec",
     "vec_to_tril_matrix",
+    # torch.export.dynamic_shapes
+    "Dim",
```
Please don't add new entries here. Just document the functions properly so that they show up on the docs website.
In this case you most likely are missing an entry in the rst file for these two

angelayi replied:
But they are in the rst files :((

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py


Differential Revision: [D51823962](https://our.internmc.facebook.com/intern/diff/D51823962)

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][1/n] Move dynamic shapes logic"

1/n of refactoring export code:
* Moved dynamic shapes/constraints/dynamic_dims logic in torch/_export/__init__.py and torch/export/__init__.py to torch/export/dynamic_shapes.py


Differential Revision: [D51823962](https://our.internmc.facebook.com/intern/diff/D51823962)

[ghstack-poisoned]

atalman commented:
@pytorchbot revert -m "github first tooling issue" -c "ghfirst"
